pkg/inject/sidecar.go

```diff
@@ -24,28 +25,32 @@ const (
 )
 
 // Sidecar adds a new container to the deployment, connecting to the given jaeger instance
-func Sidecar(dep *appsv1.Deployment, jaeger *v1alpha1.Jaeger) {
+func Sidecar(obj runtime.Object, jaeger *v1alpha1.Jaeger) {
 	deployment.NewAgent(jaeger) // we need some initialization from that, but we don't actually need the agent's instance here
 
-	if jaeger == nil || dep.Annotations[Annotation] != jaeger.Name {
-		logrus.Debugf("Skipping sidecar injection for deployment %v", dep.Name)
-	} else {
-		decorate(dep)
-		logrus.Debugf("Injecting sidecar for pod %v", dep.Name)
-		dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, container(jaeger))
+	switch o := obj.(type) {
+	case *appsv1.Deployment:
+	case *appsv1.StatefulSet:
+		if jaeger == nil || obj.Annotations[Annotation] != jaeger.Name {
```

Comment:
@jpkrohling even with type assertions here, I'm running into errors like 

```
obj.Annotations undefined (type "github.com/jaegertracing/jaeger-operator/vendor/k8s.io/apimachinery/pkg/runtime".Object has no field or method Annotations)
```

Reply:
You should use `o` instead of `obj` here :)

Reply:
I went hunting everywhere. Thanks for pointing this out!